praxis/layers/injection/fp8_nvidia_gpu.py

```diff
@@ -0,0 +1,115 @@
+# coding=utf-8
+# Copyright 2023 The Pax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Op wrappers to support FP8 GEMMs."""
+
+from functools import partial
+
+from flax.linen import fp8_ops
+
+from jax import custom_vjp
+from jax import lax
+from jax import numpy as jnp
+
+from praxis import base_layer
+from praxis import layers
+from praxis import pax_fiddle
+from praxis import pytypes
+
+WeightInit = base_layer.WeightInit
+WeightHParams = base_layer.WeightHParams
+JTensor = pytypes.JTensor
+OVERWRITE_WITH_GRADIENT = \
+    base_layer.WeightHParamsCollection.OVERWRITE_WITH_GRADIENT
+
+
+def fp8_einsum(eqn, x, k, compute_dtype, x_scale, x_amax_history, k_scale,
```

Comment:
Optional: No need for this global function. I would inline it. That increases readability because user does not have to track which variable goes where.